### PR TITLE
SoftDevice Controller: Use default TX power level set by kconfigs

### PR DESCRIPTION
**Pull request by weeTike**
Enable a default radio TX power level to be set via kconfig in
the SDC

Signed-off-by: Timothy Keys <timothy.keys@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -41,6 +41,15 @@ Protocols
...
 See `Samples`_ for lists of changes for the protocol-related samples.
 
+Bluetooth LE Controller
+-----------------------
+
+These changes are valid for :ref:`nrfxlib:softdevice_controller`.
+
+* Added:
+
+  * The SoftDevice Controller now respects ``CONFIG_BT_CTLR_TX_PWR_*``.
```
Comment (wiba-nordic) [inline doc/nrf/releases/release-notes-changelog.rst:51]:
```suggestion
* Added:

  * The :ref:`nrfxlib:softdevice_controller` now respects ``CONFIG_BT_CTLR_TX_PWR_*``.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/radio_nrf5_txp.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#define RADIO_TXP_DEFAULT -12
+#elif defined(CONFIG_BT_CTLR_TX_PWR_MINUS_16)
+#define RADIO_TXP_DEFAULT -16
+#elif defined(CONFIG_BT_CTLR_TX_PWR_MINUS_20)
+#define RADIO_TXP_DEFAULT -20
+#elif defined(CONFIG_BT_CTLR_TX_PWR_MINUS_30)
+#define RADIO_TXP_DEFAULT -30
+#elif defined(CONFIG_BT_CTLR_TX_PWR_MINUS_40)
+#define RADIO_TXP_DEFAULT -40
+#endif
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/radio_nrf5_txp.h:37]:
```suggestion
#define RADIO_TXP_DEFAULT -40
#else
#error Unknown output power
#endif
```

File: subsys/bluetooth/controller/hci_driver.c [hunk 3]
```diff
@@ -533,6 +534,13 @@ static int configure_supported_features(void)
 		}
 	}
 
+#ifdef RADIO_TXP_DEFAULT
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/hci_driver.c:537]: No need to call the API if the default output power is 0.
```suggestion
#if RADIO_TXP_DEFAULT != 0
```
I also assume that RADIO_TXP_DEFAULT will always be defined

**Review by rugeGerritsen — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/hci_driver.c [hunk 4]
```diff
@@ -533,6 +534,13 @@ static int configure_supported_features(void)
...
+	err = sdc_default_tx_power_set(RADIO_TXP_DEFAULT);
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/hci_driver.c:538]: Discussed offline. We should also add the FEM gain here so that the application can set its output power to a high value in a simple way. That will be addressed in a separate PR

**Review by rugeGerritsen — APPROVED** [review]

**Review by WilleBack — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]
Sorry, approved with wrong account previously.

File: subsys/bluetooth/controller/radio_nrf5_txp.h [hunk 5]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (carlescufi) [inline subsys/bluetooth/controller/radio_nrf5_txp.h:4]: Why is this Apache-2.0?

**Review by carlescufi — APPROVED** [review]
